### PR TITLE
Add CourtBallGameApp entity (Issue #24) 

### LINKED ISSUE #24: Need to add CourtBallGame entity

### PR DESCRIPTION
**Pull request by amitCohen2**
1. Create CourtBallGameApp app with the starting files
2. Add CourtBallGameApp model to the player.models
3. Add CourtBallGameApp to the meet_balls/settings.py

closes #24

### CODE REVIEW

**Comment by yftacherzog** [comment]
You have a bunch of files there generated by your IDE (all those .idea files and I suspect the changes to gitignore as well). They should not be pushed to git (`git rm` them and amend the commit).

**Comment by amitCohen2** [comment]
> You have a bunch of files there generated by your IDE (all those .idea files and I suspect the changes to gitignore as well). They should not be pushed to git (`git rm` them and amend the commit).

Fixed

File: CourtBallGame/models.py
Comment (Danielsio) [inline CourtBallGame/models.py:24]: Default value of blank is set to false, so you can remove it if you want to.

File: meetballs/settings.py
Comment (Danielsio) [inline meetballs/settings.py:43]: you have duplicate field 'meet_balls_app.apps.MeetBallsAppConfig',
and 
'court.apps.CourtConfig',

**Review by Danielsio — COMMENTED** [review]
Great Work so far Amit, I went through your code and left minor comments.

Also don't forget to Add in the PR descreption that it's Depends on issues/ PR that needs to be merged before this one.

**Comment by amitKaplansky** [comment]
@amitCohen2  @ZvikaNaorCohen I think you need to add to the second commit: "add court entity" - do not merge before Court entity is merged.

File: CourtBallGame/models.py
Comment (amitCohen2) [reply CourtBallGame/models.py:24]: blank=False is a parameter that can be passed to CharField to indicate that the field is required and cannot be left blank.
I think it is required here.

File: CourtBallGame/models.py
Comment (Yuval-Vino) [inline CourtBallGame/models.py:7]: This enum is already defined in `player.models`, after rebasing you should be able to import it

File: meetballs/settings.py
Comment (amitCohen2) [reply meetballs/settings.py]: Done

**Comment by yftacherzog** [comment]
@amitCohen2 please rebase on upstream/main so you only have a single commit (no merge commits inside PRs)

**Comment by yftacherzog** [comment]
Please add tests as soon as possible.

File: CourtBallGame/__init__.py
Comment (yftacherzog) [inline CourtBallGame/__init__.py:1]: You now have both `CourtBallGame` and ``court_ball_game`.

Comment (amitCohen2) [reply CourtBallGame/__init__.py:1]: Thanks for notice me.
Done.

Comment (yftacherzog) [reply CourtBallGame/__init__.py:1]: Something is broken with the tests now. and your commit is a merge-commit now for some reason.

File: court_ball_game/tests.py
Comment (Yuval-Vino) [inline court_ball_game/tests.py:12]: instead of declaring test_court in each function try using pytest.fixture

File: court_ball_game/models.py
Comment (yftacherzog) [inline court_ball_game/models.py:5]: This is not being used anywhere as far as i can see.

File: court_ball_game/models.py
Comment (yftacherzog) [inline court_ball_game/models.py:8]: Do we need the unique together stuff that we have for event player?

Comment (amitCohen2) [reply court_ball_game/models.py:8]: No, I tryed to merge t because I had some conflicts in settings.py file

Comment (yftacherzog) [reply court_ball_game/models.py:8]: Don't know if this comment belongs here, but you should solve conflicts with rebase not merge. If you need help solving the current issue send out a shout.

File: court_ball_game/tests.py
Comment (amitCohen2) [reply court_ball_game/tests.py:12]: Thanks for the tip, will be fixed.

File: court_ball_game/models.py
Comment (yftacherzog) [reply court_ball_game/models.py:6]: Back to the original comment, I think we need the unique together stuff here. Similarly to the event player module.

Comment (amitCohen2) [reply court_ball_game/models.py:6]: Done.

File: court_ball_game/tests.py
Comment (amitCohen2) [reply court_ball_game/tests.py:11]: Done

File: CourtBallGame/__init__.py
Comment (amitCohen2) [reply CourtBallGame/__init__.py:1]: Fixed.

File: court_ball_game/models.py
Comment (amitCohen2) [reply court_ball_game/models.py]: Removed.

File: CourtBallGame/models.py
Comment (amitCohen2) [reply CourtBallGame/models.py]: Got your tip, Its a default parameter.

File: court_ball_game/tests.py
Comment (yftacherzog) [reply court_ball_game/tests.py:11]: I don't see the unique_together stuff, not the test for it.

File: court_ball_game/models.py
Comment (yftacherzog) [reply court_ball_game/models.py:6]: I don't see the unique_together stuff, not the test for it.

File: court_ball_game/tests.py
Comment (yftacherzog) [reply court_ball_game/tests.py:24]: Same goes for similar cases below

File: court_ball_game/tests.py
Comment (yftacherzog) [inline court_ball_game/tests.py:50]:
```suggestion
        id = court.id
        court_ball_game = CourtBallGame.objects.create(court_id=court, ball_game=BallGame.Basketball)
        court_ball_game.delete()
        with pytest.raises(CourtBallGame.DoesNotExist):
            CourtBallGame.objects.get(id=id)
```

File: court_ball_game/models.py
Comment (yftacherzog) [inline court_ball_game/models.py:7]: Looking at the tests, I think this should be named court rather than court_id, as it contains a reference to the court rather than its id
```suggestion
    court = models.ForeignKey(Court, on_delete=models.CASCADE)
```

File: court_ball_game/tests.py
Comment (amitCohen2) [reply court_ball_game/tests.py]: Changed to another version, 
Please tell me what do you think about it.

File: court_ball_game/tests.py
Comment (amitCohen2) [reply court_ball_game/tests.py]: Fixed

File: court_ball_game/tests.py
Comment (amitCohen2) [reply court_ball_game/tests.py]: Removed

File: court_ball_game/models.py
Comment (yftacherzog) [inline court_ball_game/models.py:11]: Add unique together constraint and associated test as in event player

**Review by yftacherzog — CHANGES_REQUESTED** [review]
rebase and fix the conflict on settings.py (make sure no duplicate entries, add the unique together constraint and I think it can be merged.

File: court_ball_game/models.py
Comment (Yuval-Vino) [inline court_ball_game/models.py:15]: id is already a primary key (so its promises to be unique) also you do not need to explicitly define it better have Django default

To prevent multiple rows in the table with the same "court" and "ball game" values (but diffrent id) , we want to use "unique_together" attribute. This attribute should involve "court" and "ball game" since we want to ensure that there are no duplicate rows with the same values for these two columns.

Comment (amitCohen2) [reply court_ball_game/models.py]: Changed. 
Please check:)

File: court_ball_game/models.py
Comment (amitCohen2) [reply court_ball_game/models.py:11]: Done

File: court_ball_game/tests.py
Comment (Yuval-Vino) [inline court_ball_game/tests.py:27]: this way it will be easier to understand
```suggestion
            CourtBallGame.objects.create(court=court_ball_game_instance.court, ball_game=court_ball_game_instance.ball_game)
            
            
            
```

Comment (amitCohen2) [reply court_ball_game/tests.py]: Fixed

**Review by Yuval-Vino — APPROVED** [review]

**Pull request merged**
